src/lib.rs

```diff
@@ -405,6 +412,28 @@ impl State {
         }
     }
 
+    ///
+    /// Generates a unique and human-readable execution id based on the local date and an optional
+    /// sequence number if the original directory already exists.
+    ///
+    fn compute_execution_id(&self) -> String {
```

Comment:
Even though it looks ok here, I believe "lib" is becoming too big, wouldn't it be better if you split this logic into a new module/struct and send the necessary information?

I envision this new struct with something like:

Reports_Manager {
  current: index,
  
  fn new(output_dir: string) {..}
  
  compute_execution_id() -> String {
   i+=1;
   ...
   }
   ...etc.
 }